exceptions.AssertionError:  in addOrGetTorrentIDSReturn

Latest devel, with tunnels enabled.
```
ERROR   1432082812.87 RemoteTorrentHandler:600   failed to save data for download fbbd64b31eb3de5b546995e6f61aeb892968bc8f.torrent: info misses key pieces
ERROR   1432086838.89             log:455   Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/pouwelse/GITHUB/tribler/Tribler/Core/Utilities/twisted_thread.py", line 54, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1192, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1201, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 824, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "/home/pouwelse/GITHUB/tribler/Tribler/dispersy/community.py", line 2050, in _process_message_batch
    return self._on_batch_cache(meta, batch)
  File "/home/pouwelse/GITHUB/tribler/Tribler/dispersy/community.py", line 2092, in _on_batch_cache
    self.on_messages(messages)
  File "/home/pouwelse/GITHUB/tribler/Tribler/dispersy/community.py", line 2206, in on_messages
    if self._dispersy.store_update_forward(possibly_messages, True, True, False):
  File "/home/pouwelse/GITHUB/tribler/Tribler/dispersy/dispersy.py", line 1694, in store_update_forward
    if self._update(possibly_messages) == False:
  File "/home/pouwelse/GITHUB/tribler/Tribler/dispersy/util.py", line 151, in wrapper
    return_value = func(*args, **kargs)
  File "/home/pouwelse/GITHUB/tribler/Tribler/dispersy/dispersy.py", line 1719, in _update
    messages[0].handle_callback(messages)
  File "/home/pouwelse/GITHUB/tribler/Tribler/community/channel/community.py", line 391, in _disp_on_torrent
    self._channelcast_db.on_torrents_from_dispersy(torrentlist)
  File "/home/pouwelse/GITHUB/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 1552, in on_torrents_from_dispersy
    torrent_ids, inserted = self.torrent_db.addOrGetTorrentIDSReturn(infohashes)
  File "/home/pouwelse/GITHUB/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 503, in addOrGetTorrentIDSReturn
    assert all(torrent_id for torrent_id in torrent_ids), torrent_ids
exceptions.AssertionError: [308452, None, 25260, 308455, 308454, 308457, 308451, 308456, 308450, 308453, 308458, 308449, 308459]
```

--- a/Tribler/Core/CacheDB/SqliteCacheDBHandler.py
+++ b/Tribler/Core/CacheDB/SqliteCacheDBHandler.py
@@ -384,13 +384,16 @@ class TorrentDBHandler(BasicDBHandler):
         return self.getTorrentIDS([infohash, ]).get(infohash)
 
     def getTorrentIDS(self, infohashes):
-        to_select = []
+        to_return = {}
 
+        to_select = []
         for infohash in infohashes:
             assert isinstance(infohash, str), "INFOHASH has invalid type: %s" % type(infohash)
             assert len(infohash) == INFOHASH_LENGTH, "INFOHASH has invalid length: %d" % len(infohash)
 
-            if infohash not in self.infohash_id:
+            if infohash in self.infohash_id:
+                to_return[infohash] = self.infohash_id[infohash]
+            else:
                 to_select.append(bin2str(infohash))
 
         parameters = '?,' * len(to_select)
@@ -400,9 +403,10 @@ class TorrentDBHandler(BasicDBHandler):
         for torrent_id, infohash in torrents:
             self.infohash_id[str2bin(infohash)] = torrent_id
 
-        to_return = {}
         for infohash in infohashes:
-            to_return[infohash] = self.infohash_id.get(infohash)
+            if infohash not in to_return:
+                to_return[infohash] = self.infohash_id.get(infohash)
+        assert len(to_return) == len(infohashes)
         return to_return
 
     def getInfohash(self, torrent_id):

